plugins/modules/packaging/os/pkgin.py

```diff
@@ -255,8 +266,11 @@ def install_packages(module, packages):
     install_c = 0
 
     for package in packages:
-        if query_package(module, package):
+        query_result = query_package(module, package)
+        if query_result:
             continue
+        elif query_result is None:
+            module.fail_json(msg="failed to find package %s for installation" % package)
```

Comment:
```suggestion
        elif query_result is None:
            module.fail_json(msg="failed to find package %s for installation" % package)
```
wouldn't it be better to use `module.warn` here? I think this change can break current users pipelines when only one package from a passed list is unavailable

Reply:
@felixfontein ^ what do you think?

Reply:
I'd guess that it already didn't succeed before, but then when the install command was actually run. Exiting here allows to have a nicer message, I guess.

Reply:
Another problem is that `None` covers actually 2 error cases, one where the package could not be matched and the search returned an error `rc != 0`.